tests/run/i6996.scala

```diff
@@ -0,0 +1,14 @@
+object Test {
+
+  def isAType(arg: String): Unit = arg match {    
+    case _ : "a" => println("an `a`")
+    case _ => println("not `a`")
+  }
+
+  def main(args: Array[String]): Unit = {
+    isAType("a")
+    println(new String("a").isInstanceOf["a"])
```

Comment:
I argue in https://github.com/lampepfl/dotty/issues/6996 that this should return true.